numba/tests/test_linalg.py

```diff
@@ -1293,8 +1300,131 @@ def inner_test_loop_fn(A, dt, **kwargs):
         # checked)
         bad1D = np.array([1.], dtype=np.float64)
         bad2D = np.array([[1.], [2.], [3.]], dtype=np.float64)
-        self.assert_dimensionally_invalid(rn, cfunc, (ok, bad1D))
-        self.assert_dimensionally_invalid(rn, cfunc, (ok, bad2D))
+        self.assert_dimensionally_invalid(cfunc, (ok, bad1D))
+        self.assert_dimensionally_invalid(cfunc, (ok, bad2D))
+
+
+class TestLinalgSolve(TestLinalgSystems):
+    """
+    Tests for np.linalg.solve.
+    """
+
+    # check that a singular system raises
+    def assert_raise_on_singular(self, cfunc, args):
+        msg = "Matrix is singular to machine precision."
+        self.assert_error(cfunc, args, msg, np.linalg.LinAlgError)
+
+    @needs_lapack
+    def test_linalg_solve(self):
+        """
+        Test np.linalg.solve
+        """
+        cfunc = jit(nopython=True)(solve_system)
+
+        def check(a, b, **kwargs):
+            expected = solve_system(a, b, **kwargs)
+            got = cfunc(a, b, **kwargs)
+
+            # check that the computed results are contig and in the same way
+            self.assert_contig_sanity(got, "F")
+
+            use_reconstruction = False
+            # try plain match of the result first
+            try:
+                np.testing.assert_array_almost_equal_nulp(
+                    got, expected, nulp=10)
+            except Exception:
+                # plain match failed, test by reconstruction
```

Comment:
OK. Should we suddenly start getting failures on some platform I can always resurrect this.